config/configauth/serverauth.go

```diff
@@ -75,6 +81,10 @@ type GRPCUnaryInterceptorFunc func(ctx context.Context, req interface{}, info *g
 // See ServerAuthenticator.GRPCStreamServerInterceptor.
 type GRPCStreamInterceptorFunc func(srv interface{}, stream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler, authenticate AuthenticateFunc) error
 
+// HTTPInterceptorFunc defines the signature for the function intercepting HTTP calls, useful for authenticators to use as
+// types for internal structs, making it easier to mock them in tests.
+type HTTPInterceptorFunc func(handler http.Handler, authenticate AuthenticateFunc) http.Handler
```

Comment:
Not sure why this is necessary since you have already AuthenticateFunc. Why do you need more?